src/docs/pages/introduction/getting-started/react.md

```diff
@@ -109,6 +109,107 @@ There are a few cases where native events of Web Components do not behave as exp
   LdInput <code>onChange</code> event invokes when the component loses focus (and the value changed). This is the standard browser behavior but differs from the native React <code>onChange</code> event. Use the <code>onInput</code> event in cases you want to handle user input immediately while typing. Find additional information in the <a href="guides/event-handling/">Event handling guide</a>.
 </ld-notice>
 
+## Cookbook
+
+### React Router
+
+[React Router](https://reactrouter.com/) is a widely used library for routing in client-side react applications. This recipe shows you how to integrate Liquid Oxygen navigation components with React Router. Although the recipe is based on React Router, interacting with other libraries should look quite similar.
+
+Liquid Oxygen provides some sophisticated components for navigation. This recipe uses the following:
+
+- [LdSidenav](/components/ld-sidenav/) is the recommended navigation solution for most applications using Liquid Oxygen. It is flexible, responsive and provides multiple levels of navigation.
+- [LdBreadcrumbs](/components/ld-breadcrumbs/) component is a simple way to display the current location in your application.
+
+As this recipe focuses on how to put together Liquid Oxygen and React Router, we'll only highlight the relevant parts of the code. We assume you already have a basic React Router setup in place. If not, please refer to the [React Router documentation](https://reactrouter.com/en/start/tutorial).
+
+Let's get started with routing. We cannot use the `<Link>` component provided by React Router within LdSidenavNavitem, therefore we use the [useNavigate](https://reactrouter.com/en/main/hooks/use-navigate) hook to navigate to a different route.
+
+```tsx
+// Sidebar.tsx
+//...
+const navigate = useNavigate()
+
+const handleLink = useCallback(
+  (e: React.MouseEvent) => {
+    e.preventDefault()
+    const href = (e.target as HTMLLdSidenavNavitemElement).href
+    if (href) {
+      navigate(href)
+    }
+  },
+  [navigate]
+)
+
+return (
+  // ...
+  <LdSidenavNavitem
+    href={`/mathematical-foundations/discrete-mathematics`}
```

Comment:
I also thought about being more realistic and iterating over some "routes" but this would introduce a layer of complexity that does not support explaining using the router. Therefore I dropped looping over links.

And setting the href to `LdSidenavNavitem` is important to get an anchor rendered. 

Anyways, you are right, I do not need the object as prop. :)

Reply:
The last example that also includes "how to integrate the navigation hook properly" is rather helpful imho. Accessing the href prop inside the click handler (and having to type-cast to HTMLLdSidenavNavitem) makes it more complicated, I think.

Reply:
well good point regarding the type casting. 

I adopted this code from [Boris example](https://github.com/emdgroup-liquid/liquid/issues/466#issuecomment-1380266535) and I liked that it just reuses the href attribute as it prevents deviation. 🤔

Let me have another look.

Reply:
I am using inline handler now. Although it is not best practice, it removes some of the the overhead regarding type casting and hooks and focuses on the topic at hand.